roles/openshift_facts/library/openshift_facts.py

```diff
@@ -1187,11 +1189,12 @@ def get_defaults(self, roles):
                         iptables_sync_period='5s', set_node_ip=False)
             defaults['node'] = node
 
-        if 'nfs' in roles:
-            nfs = dict(exports_dir='/var/export', registry_volume='regvol',
-                       export_options='*(rw,sync,all_squash)')
-            defaults['nfs'] = nfs
-
+        defaults['hosted'] = dict(registry_storage_type=None,
+                                  registry_storage_host=None,
+                                  registry_storage_volume_dir='/var/exports',
+                                  registry_storage_volume='registry',
+                                  registry_storage_volume_capacity='5Gi',
+                                  registry_storage_export_options='*(rw,sync,all_squash)')
```

Comment:
Well, the inventory variables for your structure coming through `oo_openshift_env` would be

```
openshift_hosted_registry_storage_type=
openshift_hosted_registry_storage_volume_name=
openshift_hosted_registry_storage_volume_size=
openshift_hosted_registry_storage_nfs_dir=
openshift_hosted_registry_storage_nfs_options=
openshift_hosted_registry_storage_nfs_host=
openshift_hosted_registry_storage_nfs_access_modes=
openshift_hosted_registry_storage_nfs_create_pv=
```